sdk/python/feast/infra/offline_stores/file.py

```diff
@@ -215,44 +291,26 @@ def evaluate_historical_retrieval():
                         created_timestamp_column
                     ]
 
-                df_to_join.sort_values(by=right_entity_key_sort_columns, inplace=True)
-                df_to_join.drop_duplicates(
-                    right_entity_key_sort_columns,
-                    keep="last",
-                    ignore_index=True,
-                    inplace=True,
+                df_to_join = df_to_join.sort_values(by=event_timestamp_column)
```

Comment:
Wouldn't it be more consistent to run multiple sort operations one after the other? That would provide deterministic results.

Reply:
fixed